### DISCUSSION TITLE
@FocusState: focusedField binding not triggering the binding action when the focusedField is set from the Reducer

### DISCUSSION THREAD

**Discussion by pdrcabrod** [pointfreeco/swift-composable-architecture#2946] [Q&A] [answered]
Hello

First of all, congratulations on the great work you have accomplished with TCA.

I have a question. I have been playing a bit with the CaseStudies and I stumbled across a situation I do not know if it is expected or not.

In the 01-GettingStarted-FocusState case study, if I add a new handler for the binding action

```      
case .binding(\.focusedField):
          print("binding focusedField")
          return .none
```

The log will be only printed if the `focusedField` is set via tapping on the SwiftUI´s TextField. When the `focusedField` is set from the Reducer, for example if we tap in the SignIn button and we have not provided a username, the log is not printed.

Is it expected? 

I can think of a real world scenario in which something like this would be useful. ie, if we have to customize a complex custom TextField when it gains focus. How could we handle that in the binding is not being triggered?

Thank you so much in advance

**Comment by mbrandonw** [comment, marked as answer]
Hi @pdrcabrod, yeah this is to be expected. If the focus changes directly in a reducer from some _other_ action:

```swift
case .buttonTapped:
  state.focusedField = .title
```

…then the `.binding` action can't possibly be destructured in the `switch`. After all, a `.binding` action wasn't sent, the `.buttonTapped` action was sent.

If you just want to listen for any changes to `focusField` you can use the `onChange` reducer operator:

```swift
CombineReducers {
  BindingReducer()
  Reduce { state, action in … }
}
.onChange(of: \.focusField) { … }
```

You can look at the docs for `onChange` for more information of how to use it.
